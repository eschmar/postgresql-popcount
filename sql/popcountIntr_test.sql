CREATE EXTENSION popcount;

SELECT popcountIntr(15::bit(4));
SELECT popcountIntr(7::bit(4));

SELECT popcountIntr(15::bit(8));
SELECT popcountIntr(7::bit(8));

SELECT popcountIntr(15::bit(12));
SELECT popcountIntr(7::bit(12));

SELECT popcountIntr(15::bit(32));
SELECT popcountIntr(7::bit(32));

SELECT popcountIntr(30::bit(32));
SELECT popcountIntr(127::bit(32));

SELECT popcountIntr(b'10101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010');

SELECT sum(popcountIntr(x::bit(512))) from generate_series(1,1000000) x;
SELECT sum(popcountIntr(x::bit(500))) from generate_series(1,1000000) x;

SELECT popcountIntrL(15::bit(4));
SELECT popcountIntrL(7::bit(4));

SELECT popcountIntrL(15::bit(8));
SELECT popcountIntrL(7::bit(8));

SELECT popcountIntrL(15::bit(12));
SELECT popcountIntrL(7::bit(12));

SELECT popcountIntrL(15::bit(32));
SELECT popcountIntrL(7::bit(32));

SELECT popcountIntrL(30::bit(32));
SELECT popcountIntrL(127::bit(32));

SELECT popcountIntrL(b'10101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010');

SELECT sum(popcountIntrL(x::bit(512))) from generate_series(1,1000000) x;
SELECT sum(popcountIntrL(x::bit(500))) from generate_series(1,1000000) x;
