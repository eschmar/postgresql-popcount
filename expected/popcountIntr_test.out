CREATE EXTENSION popcount;
ERROR:  extension "popcount" already exists
SELECT popcountIntr(15::bit(4));
 popcountintr 
--------------
            4
(1 row)

SELECT popcountIntr(7::bit(4));
 popcountintr 
--------------
            3
(1 row)

SELECT popcountIntr(15::bit(8));
 popcountintr 
--------------
            4
(1 row)

SELECT popcountIntr(7::bit(8));
 popcountintr 
--------------
            3
(1 row)

SELECT popcountIntr(15::bit(12));
 popcountintr 
--------------
            4
(1 row)

SELECT popcountIntr(7::bit(12));
 popcountintr 
--------------
            3
(1 row)

SELECT popcountIntr(15::bit(32));
 popcountintr 
--------------
            4
(1 row)

SELECT popcountIntr(7::bit(32));
 popcountintr 
--------------
            3
(1 row)

SELECT popcountIntr(30::bit(32));
 popcountintr 
--------------
            4
(1 row)

SELECT popcountIntr(127::bit(32));
 popcountintr 
--------------
            7
(1 row)

SELECT popcountIntr(b'10101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010');
 popcountintr 
--------------
          263
(1 row)

SELECT sum(popcountIntr(x::bit(512))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

SELECT sum(popcountIntr(x::bit(500))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

SELECT popcountIntrL(15::bit(4));
 popcountintrl 
---------------
             4
(1 row)

SELECT popcountIntrL(7::bit(4));
 popcountintrl 
---------------
             3
(1 row)

SELECT popcountIntrL(15::bit(8));
 popcountintrl 
---------------
             4
(1 row)

SELECT popcountIntrL(7::bit(8));
 popcountintrl 
---------------
             3
(1 row)

SELECT popcountIntrL(15::bit(12));
 popcountintrl 
---------------
             4
(1 row)

SELECT popcountIntrL(7::bit(12));
 popcountintrl 
---------------
             3
(1 row)

SELECT popcountIntrL(15::bit(32));
 popcountintrl 
---------------
             4
(1 row)

SELECT popcountIntrL(7::bit(32));
 popcountintrl 
---------------
             3
(1 row)

SELECT popcountIntrL(30::bit(32));
 popcountintrl 
---------------
             4
(1 row)

SELECT popcountIntrL(127::bit(32));
 popcountintrl 
---------------
             7
(1 row)

SELECT popcountIntrL(b'10101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010');
 popcountintrl 
---------------
           263
(1 row)

SELECT sum(popcountIntrL(x::bit(512))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

SELECT sum(popcountIntrL(x::bit(500))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

